File: packages/shared/src/enums.ts
export const DataSource = {
  MANUAL: "MANUAL",
  GOOGLE_FIT: "GOOGLE_FIT",
  WHOOP: "WHOOP",
  MYFITNESSPAL: "MYFITNESSPAL",
  HEALTH_CONNECT: "HEALTH_CONNECT",
} as const;
export type DataSource = (typeof DataSource)[keyof typeof DataSource];

export const IntegrationProvider = {
  WHOOP: "WHOOP",
  HEALTH_CONNECT: "HEALTH_CONNECT",
} as const;
export type IntegrationProvider = (typeof IntegrationProvider)[keyof typeof IntegrationProvider];

export const SyncStatus = {
  IDLE: "IDLE",
  SYNCING: "SYNCING",
  SUCCESS: "SUCCESS",
  ERROR: "ERROR",
} as const;
export type SyncStatus = (typeof SyncStatus)[keyof typeof SyncStatus];

export const UnitSystem = {
  METRIC: "METRIC",
  IMPERIAL: "IMPERIAL",
} as const;
export type UnitSystem = (typeof UnitSystem)[keyof typeof UnitSystem];

export const OnboardingStatus = {
  PENDING: "PENDING",
  SKIPPED: "SKIPPED",
  COMPLETED: "COMPLETED",
} as const;
export type OnboardingStatus = (typeof OnboardingStatus)[keyof typeof OnboardingStatus];

export const ActivityType = {
  GYM: "GYM",
  RUNNING: "RUNNING",
  TEAM_SPORT_TRAINING: "TEAM_SPORT_TRAINING",
  MATCH: "MATCH",
  CYCLING: "CYCLING",
  WALKING: "WALKING",
  RECOVERY_SESSION: "RECOVERY_SESSION",
  OTHER: "OTHER",
} as const;
export type ActivityType = (typeof ActivityType)[keyof typeof ActivityType];

export const MatchResult = {
  WIN: "WIN",
  LOSS: "LOSS",
  DRAW: "DRAW",
} as const;
export type MatchResult = (typeof MatchResult)[keyof typeof MatchResult];

export const GoalType = {
  BODY_WEIGHT: "BODY_WEIGHT",
  CALORIE_INTAKE: "CALORIE_INTAKE",
  PROTEIN_INTAKE: "PROTEIN_INTAKE",
  TRAINING_FREQUENCY: "TRAINING_FREQUENCY",
  EXERCISE_PERFORMANCE: "EXERCISE_PERFORMANCE",
  SPORT_PERFORMANCE: "SPORT_PERFORMANCE",
  SLEEP_RECOVERY: "SLEEP_RECOVERY",
  CUSTOM: "CUSTOM",
} as const;
export type GoalType = (typeof GoalType)[keyof typeof GoalType];

export const GoalDirection = {
  INCREASE: "INCREASE",
  DECREASE: "DECREASE",
} as const;
export type GoalDirection = (typeof GoalDirection)[keyof typeof GoalDirection];

export const GoalStatus = {
  ON_TRACK: "ON_TRACK",
  AT_RISK: "AT_RISK",
  ACHIEVED: "ACHIEVED",
  PAUSED: "PAUSED",
} as const;
export type GoalStatus = (typeof GoalStatus)[keyof typeof GoalStatus];

export const ReadinessLevel = {
  HIGH: "HIGH",
  MODERATE: "MODERATE",
  LOW: "LOW",
} as const;
export type ReadinessLevel = (typeof ReadinessLevel)[keyof typeof ReadinessLevel];

export const AchievementType = {
  STREAK: "STREAK",
  PERSONAL_RECORD: "PERSONAL_RECORD",
  MILESTONE: "MILESTONE",
  WEEKLY_COMPLETION: "WEEKLY_COMPLETION",
} as const;
export type AchievementType = (typeof AchievementType)[keyof typeof AchievementType];

export const EnergyBalanceGranularity = {
  DAY: "day",
  WEEK: "week",
  MONTH: "month",
  YEAR: "year",
} as const;
export type EnergyBalanceGranularity =
  (typeof EnergyBalanceGranularity)[keyof typeof EnergyBalanceGranularity];

export const TimelineEntryKind = {
  WEIGHT: "WEIGHT",
  MEAL: "MEAL",
  TRAINING: "TRAINING",
  RECOVERY: "RECOVERY",
  ACHIEVEMENT: "ACHIEVEMENT",
} as const;
export type TimelineEntryKind = (typeof TimelineEntryKind)[keyof typeof TimelineEntryKind];

File: android/app/src/main/kotlin/com/momentum/android/network/dto/Enums.kt
package com.momentum.android.network.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

// Ported from packages/shared/src/enums.ts -- wire values must match exactly.
// Every enum here serializes by its constant name (kotlinx.serialization's
// default), which already equals the wire string for all but
// EnergyBalanceGranularity (lowercase on the wire).

@Serializable
enum class DataSource { MANUAL, GOOGLE_FIT, WHOOP, MYFITNESSPAL, HEALTH_CONNECT }

@Serializable
enum class IntegrationProvider { WHOOP, HEALTH_CONNECT }

@Serializable
enum class SyncStatus { IDLE, SYNCING, SUCCESS, ERROR }

@Serializable
enum class UnitSystem { METRIC, IMPERIAL }

@Serializable
enum class OnboardingStatus { PENDING, SKIPPED, COMPLETED }

@Serializable
enum class ActivityType {
    GYM, RUNNING, TEAM_SPORT_TRAINING, MATCH, CYCLING, WALKING, RECOVERY_SESSION, OTHER,
}

@Serializable
enum class MatchResult { WIN, LOSS, DRAW }

@Serializable
enum class GoalType {
    BODY_WEIGHT, CALORIE_INTAKE, PROTEIN_INTAKE, TRAINING_FREQUENCY,
    EXERCISE_PERFORMANCE, SPORT_PERFORMANCE, SLEEP_RECOVERY, CUSTOM,
}

@Serializable
enum class GoalDirection { INCREASE, DECREASE }

@Serializable
enum class GoalStatus { ON_TRACK, AT_RISK, ACHIEVED, PAUSED }

@Serializable
enum class ReadinessLevel { HIGH, MODERATE, LOW }

@Serializable
enum class AchievementType { STREAK, PERSONAL_RECORD, MILESTONE, WEEKLY_COMPLETION }

@Serializable
enum class MealType { BREAKFAST, LUNCH, DINNER, SNACKS }

// Retrofit's @Query params are stringified via plain .toString(), which
// would default to the Kotlin constant name (e.g. "DAY"), NOT the
// @SerialName -- that annotation only governs kotlinx.serialization's JSON
// (de)serialization. Since this enum is only ever sent as a query param
// (never inside a JSON body), the explicit wireValue property is what
// callers must actually use; @SerialName is kept too so JSON round-tripping
// stays correct if this type is ever embedded in a body later.
@Serializable
enum class EnergyBalanceGranularity(val wireValue: String) {
    @SerialName("day") DAY("day"),
    @SerialName("week") WEEK("week"),
    @SerialName("month") MONTH("month"),
    @SerialName("year") YEAR("year"),
}

@Serializable
enum class TimelineEntryKind { WEIGHT, MEAL, TRAINING, RECOVERY, ACHIEVEMENT }
